blocks/rich-text/index.js

```diff
@@ -704,9 +757,31 @@ export class RichText extends Component {
 	}
 
 	componentDidUpdate( prevProps ) {
+		if ( ! this.editor ) {
+			return;
+		}
+
+		if ( this.props.isSelected !== prevProps.isSelected ) {
+			this.toggleInsertAvailable();
+		}
+
+		if (
+			this.props.isInlineInsertionPointVisible &&
```

Comment:
Kind of seems all this functionality could be packaged in a HOC `withInlineInsertion` or something to not burden RichText so much. Not sure.

Reply:
Yes, I agree that I shouldn't burden RichText so much. I've now moved all of the inline block logic into a new `InlineBlocks` component in https://github.com/WordPress/gutenberg/pull/5794/commits/d4229ef3135d85d0079d1c6f4481deb1192ec6f0.